learners/BF_measles.Rmd

```diff
@@ -0,0 +1,884 @@
+  demography_vector = 22.67e6,  # Population size
+  # Initial conditions: Susceptible, exposed, Infectious, vaccinated, recovered
+  initial_conditions = matrix(
+    c(1 - 1 / 22.67e6, 1 / 22.67e6, 0, 0, 0),
+    nrow = 1, ncol = 5
+  )
+)
+
+# Run the SEIR model simulation
+output <- epidemics::model_default(
+  population = bfaso,
+  transmission_rate = 15 / 5,  # R0/infectious period = 15/5 = 3
+  infectiousness_rate = 1 / 8,  # 1/latent period (pre-infectious phase)
+  recovery_rate = 1 / 5,  # 1/infectious period
+  time_end = 120,  # Simulation over 120 days
+  increment = 1  # Daily increments
+)
+
+# Plot the epidemic curve for susceptible, exposed, infectious, and recovered
+output %>%
+  filter(compartment %in% c("exposed", "infectious")) %>%
+  ggplot(aes(
+    x = time,
+    y = value / 10^6,  # Scale to millions
+    col = compartment,
+    linetype = compartment
+  )) +
+  geom_line(linewidth = 1.5) +
+  theme_bw() +
+  theme(
+    legend.position = "top"
+  ) +
+  labs(
+    x = "Simulation Time (days)",
+    y = "Individuals (in millions)",
+    caption = expression(
+      paste(
+        "The dynamics of SIER model with ",
+        italic(R)[0], "= 15, ",
+        "showing the trajectories of number of exposed and infected individuals"
+      )
+    )
+  )
+```
+
+
+
+## Expected Epidemic Size & Herd Immunity
+
+Using the `{finalsize}` package implements
+[Equation 3](#eq-fs) for calculate the expected proportion
+of the population that will be infected given a specified basic
+reproduction number $(\mathcal{R}_0)$. The calculation can be done as
+follows:
+
+```{r,message=FALSE,warning=FALSE}
+# Loading the finalsize library
+library(finalsize)
+```
+
+```{r}
+finalsize::final_size(r0 = 10)
+```
+
+For $\mathcal{R}_0 \geq 10$, the results indicate that nearly the entire
+population is expected to be infected by measles. The $HIT$ needed to
+contains the epidemic is about $93\%$, uaing the formula given in
+[Equation 4](#eq-hit).
+
+# Age-Structured Model
+
+The age structure of Burkina Faso is characteristic of a young
+population, with a majority of the population being under 25 years old.
+According to recent estimates ([United Nations, 2023](https://population.un.org/wpp/); [Central Intelligence Agency, 2023](https://www.cia.gov/the-world-factbook/countries/burkina-faso/); [World Bank, 2023](https://data.worldbank.org/indicator/SP.POP.TOTL?locations=BF)), the age structure
+is broken down as follows:
+
+
+-   $[0\to15)$ years: \~ 44 (43-45)% of the population
+-   $[15\to 25)$ years: \~19.5 (19-20)%
+-   $[25\to 55)$ years: \~29 (28-30)%
+-   $[55\to 65)$ years: \~5 (3-5)%
+-   $65+$ : \~2.5 (2-3)%
+
+
+## **Contact Matrix**
+
+A contact matrix typically consists of rows and columns representing
+different age groups, where each cell indicates the average number of
+contacts between individuals of the age group corresponding to that row
+and column.
+
+Although we do not have a direct access for contact matrix for Burkina
```

Comment:
Thanks for pointing out this new development.

Reply:
I think @Bisaloo is planning this (maybe at upcoming dev day?)

Reply:
Yes, this is planned in the upcoming dev day.

(Technical aside: this will be in the contactmatrix package, which defines the new shared class for all social contact data packages, so it can benefit all packages. Even socialmixr, which currently already has this feature, would be able to simplify its code after this is implemented.)